department-of-veterans-affairs/caseflow
# Create job to cache travel board appeals immediately

Resolves #15426 

### Description

- Moved `FetchHearingLocationsForVeteransJob` logic to `GeomatchService`
- Moved `UpdateCachedAppealsAttributesJob` logic to `CachedAppealService`
  - Refactored some existing logic to get rid of some CodeClimate warnings. The remaining ones I think we can ignore
- Add `Hearings::GeomatchAndCacheAppealJob`, and start it when completing `ChangeHearingRequestTypeTask`
- Update and add tests
- Resolves merge conflicts from #15482 and #15478

### Acceptance Criteria
- [x] When a user completes the "Convert hearing to virtual" action on a Travel Board hearing, it immediately shows up in the geo-matched RO queue

### Testing Plan
1. Login as BVASYELLOW
2. Go to http://localhost:3000/queue/appeals/1986897
3. Work the change appeal type task
4. Go to the schedule veteran page for St. Petersburg, and ensure the case shows up
